src/Controls/src/Core/Handlers/Items2/iOS/LayoutFactory2.cs

```diff
@@ -396,6 +401,38 @@ public static UICollectionViewLayout CreateCarouselLayout(
 		return layout;
 	}
 #nullable enable
+
+	public static bool IsIndexPathValid(NSIndexPath indexPath, UICollectionView collectionView)
+	{
+		try
+		{
+			if (indexPath is null || collectionView is null)
+			{
+				return false;
+			}
+
+			if (indexPath.Item < 0 || indexPath.Section < 0)
+			{
+				return false;
+			}
+
+			if (indexPath.Section >= collectionView.NumberOfSections())
+			{
+				return false;
+			}
+
+			if (indexPath.Item >= collectionView.NumberOfItemsInSection(indexPath.Section))
+			{
+				return false;
+			}
+
+			return true;
+		}
+		catch (Exception ex) when (ex is ObjectDisposedException or InvalidOperationException)
+		{
+			return false;
```

Comment:
Could we log the exception for debugging purposes?

Reply:
Added the log for debugging purposes